# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ctwm/yyparse.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

*** ./gram.y~	Tue Dec 11 16:38:52 2001
--- ./gram.y	Thu Jun 13 08:42:58 2002
***************
*** 419,425 ****
  		| WINDOW_RING		{ list = &Scr->WindowRingL; }
  		  win_list
  		| WINDOW_RING           { Scr->WindowRingAll = TRUE; }
- 		;
  		| WINDOW_RING_EXCLUDE    { if (!Scr->WindowRingL)
  		                              Scr->WindowRingAll = TRUE;
  		                          list = &Scr->WindowRingExcludeL; }
--- 419,424 ----
***************
*** 427,432 ****
--- 426,432 ----
  
  		| WINDOW_GEOMETRIES 	 {  }
  		  wingeom_list
+ 		;
  
  noarg		: KEYWORD		{ if (!do_single_keyword ($1)) {
  					    twmrc_error_prefix();
***************
*** 683,691 ****
  		| wingeom_entries wingeom_entry
  		;
  
! wingeom_entry	: string string	{ AddToList (&Scr->WindowGeometries, $1, $2) }
! 
! 
  
  squeeze		: SQUEEZE_TITLE { 
  				    if (HasShape) Scr->SqueezeTitle = TRUE;
--- 683,690 ----
  		| wingeom_entries wingeom_entry
  		;
  
! wingeom_entry	: string string	{ AddToList (&Scr->WindowGeometries, $1, $2); }
! 		;
  
  squeeze		: SQUEEZE_TITLE { 
  				    if (HasShape) Scr->SqueezeTitle = TRUE;
***************
*** 834,839 ****
--- 833,839 ----
  occupy_workspc_entry	: string {
  				AddToClientsList ($1, client);
  			  }
+ 			;
  
  occupy_window_list	: LB occupy_window_entries RB {}
  			;
***************
*** 845,850 ****
--- 845,851 ----
  occupy_window_entry	: string {
  				AddToClientsList (workspace, $1);
  			  }
+ 			;
  
  icon_list	: LB icon_entries RB {}
  		;
***************
*** 962,967 ****
--- 963,970 ----
  					  RemoveDQuote(ptr);
  					  $$ = (unsigned char*)ptr;
  					}
+ 		;
+ 
  number		: NUMBER		{ $$ = $1; }
  		;
  
